src/components/Candidates/Roles/CandidatesOpenPositions/FormsModal/FormsModal.js

```diff
@@ -18,59 +20,93 @@ type Props = {
+  const [message, setMessage] = useState('Thanks for your interest! Check your Tec email.')
+  const [severity, setSeverity] = useState('success')
 
-const FormsModal = (props: Props) => {
-  let name: HTMLInputElement
-  let career: HTMLInputElement
-  let semester: HTMLInputElement
-  let matricualtionNumber: HTMLInputElement
-  let comments: HTMLInputElement
+  const handleClose = (event, reason) => {
+    if (reason !== 'clickaway') {
+      setOpen(false)
+    }
+  }
 
-  const createMail = () => {
-    const emailBody = `Hola soy ${name.value}
-      estudiante de ${career.value} de ${semester.value} semestre. \n
-      ${comments.value}`
-    const { selectedPosition } = props
+  const createMail = (recaptchaKey: string) => {
+    const emailBody = `Hola soy ${name}
+      estudiante de ${career} de ${semester} semestre. \n
+      ${comments}`
     return ({
       message: emailBody,
-      from_name: name.value,
-      reply_to: `${matricualtionNumber.value}@itesm.mx`,
+      from_name: name,
+      reply_to: `${matricualtionNumber}@itesm.mx`,
       position: selectedPosition.title,
+      recaptchaKey,
     })
   }
 
-  const handleSubmit = (event: SyntheticEvent<HTMLButtonElement>) => {
-    event.preventDefault()
-    const { onSubmit, toggle } = props
+  const handleSubmit = (recaptchaKey: string) => {
     onSubmit()
-    if (name.value && career.value && semester.value
-      && matricualtionNumber.value && comments.value) {
-      toggle()
-      const mailParams = createMail()
-      sendJoinEmail(mailParams).then((result) => {
-        if (result.status === 200) {
-          // eslint-disable-next-line no-alert
-          alert('Thanks for your interest! \nCheck your Tec email \n')
+    if (name && career && semester
+      && matricualtionNumber && comments) {
+      const mailParams = createMail(recaptchaKey)
+      setIsLoading(true)
+      sendJoinUsEmail(mailParams).then((result) => {
+        if (result.status === 1) {
+          setOpen(true)
+          setMessage('Thanks for your interest! Check your Tec email.')
+          setSeverity('success')
         } else {
-          throw new Error('Email-Server Error, Retry Later')
+          setOpen(true)
+          setMessage('Something went wrong! Please retry later')
+          setSeverity('error')
         }
+        setName('')
+        setCareer('')
+        setSemester('')
+        setMatricualtionNumber('')
+        setComments('')
       }, (error) => {
-        // eslint-disable-next-line no-alert
-        alert(`Something went wrong! \n${error.text}`)
+        setOpen(true)
+        setMessage('Something went wrong! Please retry later')
+        setSeverity('error')
+        throw new Error(`Something went wrong! \n${error.message}`)
+      }).finally(() => {
+        setIsLoading(false)
+        toggle()
       })
+    } else {
+      setOpen(true)
+      setMessage('Please complete all fields')
+      setSeverity('error')
```

Comment:
I believe required field logic should be handled by the HTML 'required 'keyword instead

Reply:
that's outside our app scope, I'll put it as a hotfix issue